lib/eth/contract/function_input.rb

```diff
@@ -26,13 +26,17 @@ class Contract::FunctionInput
     # @param data [Hash] contract abi data.
     def initialize(data)
       @raw_type = data["type"]
-      @type = Eth::Abi::Type.parse(data["type"])
+      @type = Eth::Abi::Type.parse(data["type"], data.dig("components"))
```

Comment:
Nit: No need to `dig` one level deep

```suggestion
      @type = Eth::Abi::Type.parse(data["type"], data["components"])
```